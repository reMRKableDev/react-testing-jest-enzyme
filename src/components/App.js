import React, { Component } from "react";
import "./styles/App.css";

import UserList from "./UserList";
import Header from "./Header";

import fetchApiData from "../services/fetchJSONPlaceholder";

import retrievingList from "../helpers/isRetrievingList";
import retrievingEmptyList from "../helpers/isRetrievingEmptyList";
import receivingError from "../helpers/isReceivingError";

class App extends Component {
  state = {
    name: "Malcolm",
    usersList: [],
    status: "initial"
  };

  componentDidMount() {
    this.fetchUser();
  }

  fetchUser = async () => {
    try {
      const results = await fetchApiData();
      this.setState({ usersList: results, status: "done" });
    } catch (error) {
      this.setState({ status: "error" });
    }
  };

  render() {
    const { status, usersList } = this.state;
    return (
      <div className="app">
        <header className="header">
          <Header name={this.state.name} />
        </header>
        {retrievingList(status)}
        {retrievingEmptyList(status, usersList)}
        {receivingError(status)}
        <UserList retrievedUsers={this.state.usersList} />
      </div>
    );
  }
}

export default App;
