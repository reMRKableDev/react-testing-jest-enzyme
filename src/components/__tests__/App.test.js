import React from "react";
import App from "../App";
import { shallow } from "enzyme";

const shallowComponent = () => shallow(<App />);

test("shallow rendering App component", () => {
  const wrapper = shallowComponent();
  expect(wrapper).toMatchSnapshot();
});

test(".app class exist on element", () => {
  const wrapper = shallowComponent();
  expect(wrapper.find(".app").hasClass("app")).toBe(true);
});

test(".header class exist on element", () => {
  const wrapper = shallowComponent();
  expect(wrapper.find(".header").hasClass("header")).toBe(true);
});

test("Header component isn't null value", () => {
  const wrapper = shallowComponent();
  expect(wrapper.find("Header")).not.toBeNull();
});

test("UserList component isn't null value", () => {
  const wrapper = shallowComponent();
  expect(wrapper.find("UserList")).not.toBeNull();
});
