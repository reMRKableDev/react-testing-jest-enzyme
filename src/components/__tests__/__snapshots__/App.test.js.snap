// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow rendering App component 1`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <Header
      name="Malcolm"
    />
  </header>
  <h2>
    Retrieving list from API
  </h2>
  <UserList
    retrievedUsers={Array []}
  />
</div>
`;
